src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.iOS.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace System.IO
+{
+    public static partial class Path
+    {
+        private static string s_defaultTempPath = string.Empty;
+
+    private static string DefaultTempPath =>
+        s_defaultTempPath ?? (s_defaultTempPath = Interop.Sys.SearchPathTempDirectory()) ??
+        throw new System.Runtime.InteropServices.ExternalException();
```

Comment:
I think we should throw InvalidOperationException instead since ExternalException is about COM interop, and we can use ThrowHelper:

```suggestion
        ThrowInvalidOperationException();
```